### PR TITLE
Disallow CRUD operations on knp prefixed Network Policy

### PR DESCRIPTION
**Pull request by robbrockbank**
## Todos
- [ ] Tests
- [ ] Documentation
- [ ] Release note

## Release Note

```release-note
None required
```

### CODE REVIEW

**Comment by caseydavenport** [comment]
I thought @doublek had already done this? https://github.com/projectcalico/calicoctl/pull/1709/files

Did it just not get included in the v3.0 branch?

**Comment by robbrockbank** [comment]
Hmmm, that's odd - dammit.  Let me check.

**Comment by robbrockbank** [comment]
@caseydavenport:  Yeah that is annoying - I didn't spot that!  Yeah it looks like it wasn't included which is a little sad.

File: calicoctl/resourcemgr/networkpolicy.go [hunk 1]
```diff
@@ -39,14 +43,35 @@ func init() {
 		},
 		func(ctx context.Context, client client.Interface, resource ResourceObject) (ResourceObject, error) {
 			r := resource.(*api.NetworkPolicy)
+			if strings.HasPrefix(r.Name, conversion.K8sNetworkPolicyNamePrefix) {
+				return nil, cerrors.ErrorOperationNotSupported{
+					Operation: "create or apply",
+					Identifier: fmt.Sprintf("GlobalNetworkPolicy(%s/%s)"),
+					Reason: "NetworkPolicy resources with 'knp.' prefixed names cannot be modified using calicoctl",
```
Comment (caseydavenport) [inline calicoctl/resourcemgr/networkpolicy.go]: A better reason might be "Kubernetes network policies cannot be managed using calicoctl" or something.

Maybe "Kubernetes network policies must be managed through the kubernetes API"?

Comment (caseydavenport) [reply calicoctl/resourcemgr/networkpolicy.go]: In either case, "modified" doesn't make sense in the "create" function.

File: calicoctl/resourcemgr/networkpolicy.go [hunk 2]
```diff
@@ -39,14 +43,35 @@ func init() {
 		},
 		func(ctx context.Context, client client.Interface, resource ResourceObject) (ResourceObject, error) {
 			r := resource.(*api.NetworkPolicy)
+			if strings.HasPrefix(r.Name, conversion.K8sNetworkPolicyNamePrefix) {
+				return nil, cerrors.ErrorOperationNotSupported{
+					Operation: "create or apply",
+					Identifier: fmt.Sprintf("GlobalNetworkPolicy(%s/%s)"),
```
Comment (caseydavenport) [inline calicoctl/resourcemgr/networkpolicy.go]: Shouldn't this be `NetworkPolicy`?

Comment (caseydavenport) [reply calicoctl/resourcemgr/networkpolicy.go]: Or just `resource`?

Comment (robbrockbank) [reply calicoctl/resourcemgr/networkpolicy.go]: d'oh- ta!

**Pull request merged**